Week3/homework/6-step3.js

```diff
@@ -4,3 +4,14 @@ const arr2d = [[1, 2], [3, 4], [5, 6]];
 const arr3d = [[[1, 2], [3, 4]], [[5, 6], [7, 8]]];
 
 // add your solution here, or add a comment on how you would tackle this problem
+// this solution is only works for arr3d
+function printArray(a) {
+    for (var i = 0; i < a.length; i++) {
+        for (var z = 0; z < a[i].length; z++) {
+            for (var y = 0; y < a[i][z].length; y++)
+                console.log(a[i][z][y]);
+        }
+    }
+}
+
+printArray(arr3d)
```

Comment:
This will not work for arr2d.
It would also not work for something like:
```
const arrNd = [[[[1, 2]], [3, 4], [[[5, 6], [7, 8]], [9, 10]], [11, 12]], [[[[[[13, 14]]]]]]];
```
I would suggest reading more up on `recursion`.
What you really want is something that will handle any level of depth of nesting.  Here is an example using for loops, and then calling itself.
```
function printAllTheItems(arr){ 
  let values = [];
  for (let i = 0; i < arr.length; i++){
    if (Array.isArray(arr[i])) {
      values = values.concat(printAllTheItems(arr[i]));
    } else {
      values = values.concat([arr[i]]);
    }
  }
  return values;
};

console.log(printAllTheItems(arr2d));
console.log(printAllTheItems(arr3d));
console.log(printAllTheItems(arrNd));
```
Fewer and shorter lines of code, and handles just about anything.

When I first tried to solve this problem I used map, and in the `class_15` slack group Jim has posted a for-loop version of a much more generic utility called flatten, I would suggest having a look.
This is a version using map:
```
const runArray = (arr) => {
  arr.map((elem) => {
    if (Array.isArray(elem)) {
      return runArray(elem);
    } else {
      console.log(elem);
    }
  });
};
```
I'm not that happy with it, because, unlike Jim's `flattenArray`, it is not reusable.